community/kernel/src/main/java/org/neo4j/kernel/impl/core/NodeProxy.java

```diff
@@ -579,6 +569,17 @@ public boolean hasProperty( String key )
         }
     }
 
+    private KernelTransaction safeAcquireTransaction()
+    {
+        KernelTransaction transaction = actions.kernelTransaction();
+        Optional<Status> terminationReason = transaction.getReasonIfTerminated();
```

Comment:
Checking `transaction.isTerminated` first, and also getting the reason if true is faster in the happy path.